mobile/src/main/java/com/joaquimley/avenging/ui/list/ListFragment.java

```diff
@@ -316,6 +318,12 @@ public boolean onQueryTextChange(String queryText) {
     }
 
     @Override
+    public void onDestroyView() {
+        mCharactersRecycler.setAdapter(null);
```

Comment:
You are right! I was being lazy stackoverflow copy paster. There is a [stackoverflow post](http://stackoverflow.com/questions/30132643/recyclerview-doesnt-unregister-itself-from-the-adapter-on-orientation-change) describes our problem. If I get it right,  when setRetainInstance(true) is used, onDestroy is not called, RecyclerView doesn't detach its adapter and it leaks the memory.  Also, there is this [answer](http://stackoverflow.com/a/22884377/1463542) about usage of setRetainInstance(true) with UI fragment. So let's keep this open until we figure it out how to handle configuration changes properly.